src/async-wrap.cc

```diff
@@ -181,6 +183,33 @@ static void Initialize(Local<Object> target,
 }
 
 
+void AsyncWrap::DestroyIdsCb(uv_idle_t* handle) {
+  Environment* env = Environment::from_destroy_ids_idle_handle(handle);
```

Comment:
Context::Scope doesn't strictly need a HandleScope because it doesn't really do anything with handles, it just changes the active context.  (The context itself is a handle, of course.)

Something similar applies to TryCatch, it merely changes the active landing pad for JS exceptions.